subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/beans/BeanPropertyReader.kt

```diff
@@ -46,7 +47,9 @@ class BeanPropertyReader(
     val instantiationScheme: InstantiationScheme = instantiatorFactory.decorateScheme()
 
     private
-    val fieldSetters = relevantStateOf(beanType).map { Pair(it.name, setterFor(it)) }
+    val fieldSetters = relevantStateOf(beanType).map {
+        FieldSetter(it.field, it.unsupportedFieldType, setterFor(it.field))
```

Comment:
:x: Could capture `it.field.name` here instead of `it.field` as the field reference is only used to for the name now.

Reply:
👍 done